pkg/plugin/implementation/reqpreprocessor/reqpreprocessor.go

```
@@ -49,11 +49,14 @@ func NewUUIDSetter(cfg *Config) (func(http.Handler) http.Handler, error) {
 			case "bpp":
 				subID = req.Context["bpp_id"]
 			}
-			ctx := context.WithValue(r.Context(), subscriberIDKey, subID)
+			if subID != nil {
+				log.Debugf(ctx, "adding subscriberId to request:%s, %v", subscriberIDKey, subID)
+				ctx = context.WithValue(ctx, subscriberIDKey, subID)
```

Add a todo here to add  a ContextKey type in model, and use it here.
Till then we will ignore this error